### ISSUE TITLE
lint/suspicious/noConsole not reported if passed as function reference

### ISSUE THREAD

**Issue by JulienZD** [biomejs/biome#5620]
### Environment information

```bash
main / playground / 2.0.0-beta.1 / beta playground
```

### Rule name

lint/suspicious/noConsole

### Playground link

https://next.biomejs.dev/playground/?lintRules=all&code=bwBmACgAKQAuAHAAaQBwAGUAKAAKACAAIAB0AGEAcAAoAGMAbwBuAHMAbwBsAGUALgBsAG8AZwApAAoAKQA7AAoACgBQAHIAbwBtAGkAcwBlAC4AcgBlAGoAZQBjAHQAKAApAAoAIAAgAC4AYwBhAHQAYwBoACgAYwBvAG4AcwBvAGwAZQAuAGUAcgByAG8AcgApADsACgAKAFAAcgBvAG0AaQBzAGUALgByAGUAcwBvAGwAdgBlACgAKQAKACAAIAAuAHQAaABlAG4AKABjAG8AbgBzAG8AbABlAC4AbABvAGcAKQA7AAoA

### Expected result

Passing any of `console.log`, `console.error`, `console.debug`, etc. as a function reference to be reported.

It's common for me to use `tap(console.log)` with RxJS to debug observable streams.

For reference, here's an [ESLint playground](https://eslint.org/play/#eyJ0ZXh0Ijoib2YoKS5waXBlKFxuICB0YXAoY29uc29sZS5sb2cpXG4pO1xuXG5Qcm9taXNlLnJlamVjdCgpXG4gIC5jYXRjaChjb25zb2xlLmVycm9yKTtcblxuUHJvbWlzZS5yZXNvbHZlKClcbiAgLnRoZW4oY29uc29sZS5sb2cpO1xuIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby1jb25zb2xlIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjIwMjQsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19) where this does get reported.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by mdevils** [comment]
Here is a PR with a fix: #5648.
